backend/app_fastapi_v2.py

```
@@ -61,10 +89,30 @@ async def api_generate_review(request: ReviewRequestV2, current_user: dict = Dep
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
 
+@v2.get("/reviews", response_model=List[Review])
+async def get_past_reviews(current_user: dict = Depends(get_current_user)):
+    """Fetch all past reviews for the authenticated user."""
+    try:
+        reviews = await get_reviews_by_user(current_user["id"])
+        return reviews
+    except Exception as e:
+        logger.error(f"Error fetching reviews for user {current_user['id']}: {e}")
+        raise HTTPException(status_code=500, detail="Failed to fetch reviews.")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)